contracts/AcceleratingDistributor.sol

```diff
@@ -182,26 +182,24 @@ contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
      * @param amount The amount of the token to stake.
      */
     function stake(address stakedToken, uint256 amount) external nonReentrant onlyEnabled(stakedToken) {
-        _updateReward(stakedToken, msg.sender);
-
-        UserDeposit storage userDeposit = stakingTokens[stakedToken].stakingBalances[msg.sender];
-
-        uint256 averageDepositTime = getAverageDepositTimePostDeposit(stakedToken, msg.sender, amount);
-
-        userDeposit.averageDepositTime = averageDepositTime;
-        userDeposit.cumulativeBalance += amount;
-        stakingTokens[stakedToken].cumulativeStaked += amount;
-
-        IERC20(stakedToken).safeTransferFrom(msg.sender, address(this), amount);
+        _stake(stakedToken, amount, msg.sender);
+    }
 
-        emit Stake(
-            stakedToken,
-            msg.sender,
-            amount,
-            averageDepositTime,
-            userDeposit.cumulativeBalance,
-            stakingTokens[stakedToken].cumulativeStaked
-        );
+    /**
+     * @notice Stake tokens for rewards on behalf of `beneficiary`.
+     * @dev The caller of this function must approve this contract to spend amount of stakedToken.
+     * @dev The caller of this function is effectively donating their tokens to the beneficiary. The beneficiary
+     * can then unstake or claim rewards as they wish.
+     * @param beneficiary User that caller wants to stake on behalf of.
+     * @param stakedToken The address of the token to stake.
+     * @param amount The amount of the token to stake.
+     */
+    function stakeFor(
+        address beneficiary,
+        address stakedToken,
+        uint256 amount
+    ) external nonReentrant onlyEnabled(stakedToken) {
+        _stake(stakedToken, amount, beneficiary);
```

Comment:
tiny nit: any reason the args are in a different order in stakeFor vs _stake?

Note: no need to change if there is some reason, just a random thing I noticed.

Reply:
how are they in different order? Because `beneficiary` is first?